workshop/content/exercises/04-remediations.md

```diff
@@ -1,9 +1,391 @@
+    -p '{"spec":{"apply":true}}' --type=merge
+complianceremediation.compliance.openshift.io/rhcos4-e8-worker-sysctl-kernel-dmesg-restrict patched
+```
+
+We can verify that the compliance-operator has detected this change by re-fetching the remediation object:
+
+```
+$ oc get complianceremediations rhcos4-e8-worker-sysctl-kernel-dmesg-restrict 
+NAME                                            STATE
+rhcos4-e8-worker-sysctl-kernel-dmesg-restrict   Applied
+```
+
+So... What's going on?
+
+Given that the remediation's object is a **MachineConfig** object, the compliance-operator will
+create this object in the cluster. Let's take a look at what the operator did:
+
+```
+$ oc get machineconfigs
+NAME                                                        GENERATEDBYCONTROLLER                      IGNITIONVERSION   AGE
+00-master                                                   41d29dde06d6c72fde07b8c613743c0feb703b92   2.2.0             76m
+...
+75-rhcos4-e8-worker-periodic-e8
+...
+rendered-worker-e0b9f65ad061103ad45716abf62b5f46            41d29dde06d6c72fde07b8c613743c0feb703b92   2.2.0             2m12s
+```
+
+The output has been trimmed for readability. But the main thing to note is that a MachineConfig called
+`75-rhcos4-e8-worker-periodic-e8` was created. If we inspect it we'll notice that it has the same
+contents as the remediation that we just applied.
+
+```
+$ oc get machineconfig 75-rhcos4-e8-worker-periodic-e8 -o yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  annotations:
+    remediation/rhcos4-e8-worker-sysctl-kernel-dmesg-restrict: "2"
+  creationTimestamp: "2020-09-08T12:31:07Z"
+  generation: 1
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 75-rhcos4-e8-worker-periodic-e8
+  resourceVersion: "41724"
+  selfLink: /apis/machineconfiguration.openshift.io/v1/machineconfigs/75-rhcos4-e8-worker-periodic-e8
+  uid: 85feacdb-d86f-40b7-8b09-f38e3ff4226f
+spec:
+  config:
+    ignition:
+      version: 2.2.0
+    storage:
+      files:
+      - contents:
+          source: data:,kernel.dmesg_restrict%3D1
+        filesystem: root
+        mode: 420
+        path: /etc/sysctl.d/75-sysctl_kernel_dmesg_restrict.conf
+  fips: false
+  kernelArguments: []
+  kernelType: default
+  osImageURL: ""
+```
+
+Now we can just let the **machine-config-operator** do its job and apply this configuration
+to all of the hosts whose **MachineConfigPool** matches the `worker` role.
+
+You might have noticed that the **machine-config-operator** did a rolling restart of the nodes in
+order to apply the configuration. This is normal and expected. However, this might be problematic
+if you're applying several MachineConfig object changes, as each change will trigger a restart of
+all the nodes in that pool.
+
+Let's wait for the `worker` **MachineConfigPool** to finish updating before continuing to the next
+section.
```

Comment:
We might want to suggest how, e.g. `oc get mcp -w` or even `watch oc get mcp`

Reply:
Done.